### PR TITLE
Modify docker files to work with existing genesis.json

### PR DESCRIPTION
**Pull request by jbowen93**
This modifies the the `init.sh` script which is run by the docker `entrypoint.sh` so that it only creates a new genesis file if an existing one isn't provided. It looks for a genesis file at either `/.ethermintd/config/genesis.json` or `$HOME_DIR/config/genesis.json` if the `HOME_DIR` env var is set. 

resolves: https://github.com/celestiaorg/ethermint/issues/43

### CODE REVIEW

File: docker/Dockerfile [hunk 1]
```diff
@@ -27,6 +27,7 @@ WORKDIR /root
 
 COPY docker/entrypoint.sh .
 COPY init.sh .
+COPY docker/priv_validator_state.json /ethermint/data/priv_validator_state.json
```
Comment (adlerjohn) [inline docker/Dockerfile:30]: What's the rationale for this change in the context of the PR?

File: docker/entrypoint-debug.sh [hunk 2]
```diff
@@ -2,9 +2,13 @@
 
 set -e
 
-if [ "$1" = 'dlv' ]; then
+FILE=/ethermint/config/genesis.json
```
Comment (adlerjohn) [inline docker/entrypoint-debug.sh:5]: Seems like a leaky abstraction to hard-code the file here, instead of having some common file also usable by `init.sh`.

File: docker/priv_validator_state.json [hunk 3]
```diff
@@ -0,0 +1,5 @@
+{
```
Comment (Manav-Aggarwal) [inline docker/priv_validator_state.json:1]: What's the context of this file?

File: docker/Dockerfile (hunk 1 above)
Comment (jbowen93) [reply docker/Dockerfile:30]: Since we aren't running a `ethermintd init` command we don't setup all the standard files so we need to ensure they already exist. Since this file gets updated by the docker image we don't want to copy it in as a volume. This is the admittedly somewhat hacky workaround.

File: docker/entrypoint-debug.sh (hunk 2 above)
Comment (jbowen93) [reply docker/entrypoint-debug.sh:5]: I'm not opposed to moving this check into the `init.sh` script and having it no op if there's an initial genesis file, otherwise generate one.

Comment (adlerjohn) [reply docker/entrypoint-debug.sh:5]: Would prefer that, but not necessarily blocking

**Comment by jbowen93** [comment]
The docker build/test CI isn't going to pass because the DALC doesn't work with the newest optimint. Tracking issue https://github.com/celestiaorg/optimint/issues/383

File: init.sh [hunk 4]
```diff
@@ -9,73 +9,90 @@ LOGLEVEL="info"
...
+  # Change parameter token denominations to aphoton
+  cat $HOME/.ethermintd/config/genesis.json | jq '.app_state["staking"]["params"]["bond_denom"]="aphoton"' > $HOME/.ethermintd/config/tmp_genesis.json && mv $HOME/.ethermintd/config/tmp_genesis.json $HOME/.ethermintd/config/genesis.json
+  cat $HOME/.ethermintd/config/genesis.json | jq '.app_state["crisis"]["constant_fee"]["denom"]="aphoton"' > $HOME/.ethermintd/config/tmp_genesis.json && mv $HOME/.ethermintd/config/tmp_genesis.json $HOME/.ethermintd/config/genesis.json
+  cat $HOME/.ethermintd/config/genesis.json | jq '.app_state["gov"]["deposit_params"]["min_deposit"][0]["denom"]="aphoton"' > $HOME/.ethermintd/config/tmp_genesis.json && mv $HOME/.ethermintd/config/tmp_genesis.json $HOME/.ethermintd/config/genesis.json
+  cat $HOME/.ethermintd/config/genesis.json | jq '.app_state["mint"]["params"]["mint_denom"]="aphoton"' > $HOME/.ethermintd/config/tmp_genesis.json && mv $HOME/.ethermintd/config/tmp_genesis.json $HOME/.ethermintd/config/genesis.json
```
Comment (adlerjohn) [inline init.sh:37]: Don't these all overwrite `$HOME/.ethermintd/config/genesis.json`? If not, `$HOME/.ethermintd/config` should be abstracted into a constant. Probably should either way actually!

File: init.sh [hunk 5]
```diff
@@ -9,73 +9,90 @@ LOGLEVEL="info"
...
+    else
+      sed -i 's/create_empty_blocks = true/create_empty_blocks = false/g' $HOME/.ethermintd/config/config.toml
   fi
-fi
 
-# Allocate genesis accounts (cosmos formatted addresses)
-ethermintd add-genesis-account $KEY 100000000000000000000000000aphoton --keyring-backend $KEYRING
+  if [[ $1 == "pending" ]]; then
+    if [[ "$OSTYPE" == "darwin"* ]]; then
+        sed -i '' 's/create_empty_blocks_interval = "0s"/create_empty_blocks_interval = "30s"/g' $HOME/.ethermintd/config/config.toml
```
Comment (adlerjohn) [inline init.sh:54]: Hmm are you sure there isn't an extra `''` here after the `-i`?

File: init.sh [hunk 6]
```diff
@@ -9,73 +9,90 @@ LOGLEVEL="info"
 # to trace evm
 TRACE="--trace"
 # TRACE=""
+DIR="${HOME_DIR:-/.ethermintd}"
```
Comment (adlerjohn) [inline init.sh:12]: This sets `DIR` to `/.ethermintd` if `HOME_DIR` is not set. Are you use it's correct to use the root directory here?

File: init.sh (hunk 5 above)
Comment (jbowen93) [reply init.sh:54]: https://stackoverflow.com/questions/4247068/sed-command-with-i-option-failing-on-mac-but-works-on-linux

File: init.sh (hunk 4 above)
Comment (jbowen93) [reply init.sh:37]: It does, but this is just a copy paste from https://github.com/tharsis/ethermint/blob/main/init.sh

File: init.sh (hunk 6 above)
Comment (jbowen93) [reply init.sh:12]: This was the desired behavior. But running through the code I realize I would have to pass the `--home $DIR` flag to every `ethermintd` command. I'm going to explicitly run 2 different `ethermintd validate-genesis` commands, one that takes a file arg if the genesis file exists already, and one that uses the default home for ethermintd (which is /.ethermintd).

File: init.sh (hunk 5 above)
Comment (adlerjohn) [reply init.sh:54]: Not the extra space, the extra `''` (empty single quotes).

File: init.sh (hunk 4 above)
Comment (adlerjohn) [reply init.sh:37]: Ah I get it now actually: it's making one change to the genesis JSON at a time.

File: init.sh (hunk 5 above)
Comment (adlerjohn) [reply init.sh:54]: Ah I see now, it's the backup.

**Review by adlerjohn — APPROVED** [review]
utACK
